### ISSUE TITLE
test against live gh actions on boilerplate repos

### ISSUE THREAD

**Issue by maxheld83** [maxheld83/ghactions#22]
this would be an alternative / addition to #14.

This could be implemented as a "simple" testthat unit test.

1. There exist a bunch of boilerplate projects *inside* of the ghaction repo. (We can use the commonly used starter projects from RStudio for example).
2. Once the test gets run:
  - the relevant boilerplate projects gets pushed to a (new?) repo on github.com (this might be possible inside of R using the git wrappers from usethis.
  - *with* the currently templated version of `.github/main.workflow`.
  - then the test waits for the github api (via gh::gh) to signal that the repo in question has successfully run through the action (or actually has failed on some of them, that we might also want to test!).
  - then the test returns the appropriate condition.

The *same* test gets run on every commit inside of GitHub actions (meta), though that might require some more git auth.

Also, the tests would have to run in parallel, otherwise this would take too long.

Aaaand we should ask github first whether this violates their TOS, because we're kinda programmatically querying github actions.

**Comment by maxheld83** [comment]
this could be a problem/limitation:

> An action can't trigger other workflows. For example, a push, deployment, or any task performed within an action with the provided GITHUB_TOKEN will not trigger a workflow listening on push, deploy, or any other supported action triggers.

source: https://developer.github.com/actions/creating-workflows/workflow-configuration-options/

Will probably have to add another `GH_PAT` or something.

Makes sense, that they won't do this (ever?), but makes me thing whether this is the right approach.

**Comment by maxheld83** [comment]
@jimhester also suggested the r-appveyor project as a source for boilerplate pkg repos.

**Comment by jimhester** [comment]
In particular https://github.com/krlmlr/r-appveyor/tree/master/r-travis/fakepackage

**Comment by maxheld83** [comment]
and https://github.com/r-hub/test-packages
